www/class/centreon-clapi/centreonDependency.class.php

```diff
@@ -692,12 +706,19 @@ protected function addRelations($parameters, $relType = 'parent')
     protected function addHostgroupRelations($depId, $objectToInsert, $relType)
     {
         $table = "dependency_hostgroup" . ucfirst($relType) . "_relation";
-        $sql = "INSERT INTO {$table} (dependency_dep_id, hostgroup_hg_id) VALUES (?, ?)";
         $obj = new \Centreon_Object_Host_Group($this->dependencyInjector);
         $ids = $obj->getIdByParameter($obj->getUniqueLabelField(), array($objectToInsert));
         if (!count($ids)) {
             throw new CentreonClapiException(sprintf('Could not find host group %s', $objectToInsert));
         }
+        $dataField = ['dependency_dep_id' => $depId, 'hostgroup_hg_id' => $ids[0]];
+        if ($this->isExistingDependency($table, $dataField)) {
+            $depName = $this->getDependencyName($depId);
+            throw new CentreonClapiException(
+                sprintf('Hostgroup %s already link to the dependency %s', $objectToInsert, $depName)
+            );
+        }
+        $sql = "INSERT INTO {$table} (dependency_dep_id, hostgroup_hg_id) VALUES (?, ?)";
         $this->db->query($sql, array($depId, $ids[0]));
```

Comment:
why not using a prepare / bind / execute statement ?

Reply:
If you are not using the prepare request, please cast values for more security.
`$this->db->query($sql, array((int) $depId, (int) $ids[0]));`